src/core/agent/services/signifyNotificationService.ts

```diff
@@ -438,6 +438,27 @@ class SignifyNotificationService extends AgentService {
               });
               break;
             }
+            case OperationPendingRecordType.Exchange: {
+              const admitExchange = await this.props.signifyClient
+                .exchanges()
+                .get(operation.metadata?.said);
+              if (admitExchange.exn.r === NotificationRoute.IpexAdmit) {
+                const grantExchange = await this.props.signifyClient
+                  .exchanges()
+                  .get(admitExchange.exn.p);
+                const credentialId = grantExchange.exn.e.acdc.d;
+                if (credentialId) {
+                  await Agent.agent.ipexCommunications.waitAndUpdateCredential(
+                    credentialId
+                  );
+                }
+              }
+              callback({
+                opType: pendingOperation.recordType,
+                oid: recordId,
```

Comment:
There is an acceptance criteria to update the UI. So we need to update Redux - get the credential and mark it as `pending: false`. See `signifyOperationStateChangeHandler` in `AppWrapper.tsx`.

The only thing is `oid: recordId` here is the the exchange message SAID and not the credential SAID.

Maybe we should have `ExchangeReceiveCredential = "exchange.receivecredential"` (sorry, my previous review said not to do this but I missed this) and `oid` can be the credentialId

Reply:
In `markAcdcComplete`, we already had `this.props.eventService.emit` to update the UI. So we will remove this, right?

Reply:
Ah, my mistake. I forgot this was already done, this code is quite old. OK, in this case, we can completely remove the `callback` call here.

I will create another ticket to review if we should use `.emit` everywhere, or `callback` everywhere etc.

So just remove `callback` for now here